Lib/test/test_typing.py

```diff
@@ -580,6 +580,148 @@ def test_constructor(self):
         self.assertIs(T.__bound__, None)
 
 
+class TypeParameterDefaultsTests(BaseTestCase):
```

Comment:
You might want to test type parameters without defaults following parameters with defaults.

Reply:
There is a test for that, `test_erroneous_generic`.

Reply:
Ah, actually I was wondering whether you wanted to test a bit more precisely your error messages (I suspected that this one should do the job but it's a huge assertRaises so I wondered whether it covered everything).